databases/sqlite/plugins.nix

```diff
@@ -0,0 +1,132 @@
+{ lib
+, callPackage
+, fetchFromGitHub
+, fetchzip
+}:
+
+let
+
+  mkSqliteExt =
+    { name
+    , sourcesFn
+    , outFn
+    }:
+
+    { lib
+    , sqlite
+    , stdenv
+    }:
+    let
+      outFile = outFn (if stdenv.isDarwin then "dylib" else "so");
```

Comment:
Rather than an `outFn` I should have an optional name that defaults to the `name`.